bin/eg-generator.js

```diff
@@ -8,7 +8,14 @@ module.exports = class EgGenerator extends Generator {
     this._configuration = null;
     this.eg = this.env.eg;
     this.argv = this.env.argv;
-    this.admin = require('../lib/admin')(config.gatewayConfig.admin);
+    if (this.argv) {
+      let cliConfig = config.systemConfig.cli || {};
```

Comment:
Could we look for something like an `EG_ADMIN_URL` environment variable and fall-back to `config.gatewayConfig.admin` if it doesn't exist?